### ISSUE TITLE
Do not add the stop line when stopOnEntry is false.

### ISSUE THREAD

**Issue by TwitchBronBron** [TwitchBronBron/vscode-brightscript-language#31]
When launching a debug session with `stopOnEntry` set to `false`, is there a way to NOT add the `stop` entry in the source code. Currently the debugger watches for that initial stop so it knows when to start listening for error events, breakpoints, etc. 

However, there may be another way that truly avoids that stop entry alltogether, while still accomplishing the same goal.

**Comment by TwitchBronBron** [comment]
The main reason for needing that entry breakpoint is [this line](https://github.com/TwitchBronBron/vscode-brightscript-language/blob/f425536b9472df7eb7a1e402061a28c654c23eb1/src/RokuAdapter.ts#L161) in the rokuAdapter. 

One possible solution is to use a special print line that is unlikely to collide with the user prints...something like `"____brightscriptDebugEntryHasRun___"`.

**Comment by triwav** [comment]
I guess my main question is I'm still confused of why we need this. You mentioned in the slack 

> I think the deep link code uses it also so it knows when the app has been published the first time, so it can kill the app and relaunch with the deep link params. A special print might work in than scenario, but you could end up with 300 lines of the main function executing before the extension finally catches up.

Thinking more about it, I don't even know if we need a special print. I feel we could likely use one of the following statements on startup to know we're running (most likely one of the last 2):
```01-10 05:08:17.407 [scrpt.ctx.cmpl.time] Compiled '<APP-NAME>', id 'dev' in 185 milliseconds
01-10 05:08:17.410 [scrpt.proc.mkup.time] Processed markup dev '<APP-NAME>' in 0 milliseconds
01-10 05:08:17.418 [ui.frm.plugin.running.enter] Entering PLUGIN_RUNNING for dev
01-10 05:08:19.246 [scrpt.ctx.run.enter] UI: Entering '<APP-NAME>', id 'dev'```

Keep in mind too, I'm perfectly fine allowing the existing behavior by default. What I'd like is for the `stopOnEntry` when set to false to not enter a stop at all vs right now it just stops and then continues.

**Comment by TwitchBronBron** [comment]
Yeah, something like this would probably be enough information to detect the app has started. I'll mess around with this in the next few days and see what I come up with.

**Comment by TwitchBronBron** [comment]
@triwav I just pushed a commit to master that I think solves this issue for you. Can you pull it down and take a look?  Also note, I have another commit in master where I upgraded roku-deploy, so if you see something strange with your files being packaged, just temporarily downgrade roku-deploy in package.json and it should work as expected.

**Comment by triwav** [comment]
@TwitchBronBron appears to be working great. I'd recommend getting some other people's input but in my testing this is a definite improvement. This will likely push me to help build in support for modifying the build number to the current date time so I can use your plugin directly for clients that don't have a dedicated build script. Currently I still use Atom for that one feature and will be nice to not have to do that going forward.

**Comment by TwitchBronBron** [comment]
That's great news! @georgejecook @bvisin would you be willing to pull down master and verify as well?

**Comment by bvisin** [comment]
@TwitchBronBron and @triwav Awesome! Ya, this works great.  Best part about this is I can relaunch the channel with the remote or with the debugger not attached and it won't stop on startup!  Also, deep linking now works!  You guys ROCK!

@georgejecook give it a shot

@TwitchBronBron I upgraded `roku-deploy` and all seems fine.

**Comment by georgejecook** [comment]
Mostly good here;
however, I can’t get any output when connecting with vscode debugger (port
4711) with this build.

That’s going to hamper further debugging work, so you might want to look
into fixing that in the short-term - I don’t deem it a blocker though.

Other than that though - looks good.


On January 11, 2019 at 7:00:58 PM, Bronley Plumb (notifications@github.com)
wrote:

That's great news! @georgejecook <https://github.com/georgejecook> @bvisin
<https://github.com/bvisin> would you be willing to pull down master and
verify as well?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/TwitchBronBron/vscode-brightscript-language/issues/31#issuecomment-453695696>,
or mute the thread

**Comment by TwitchBronBron** [comment]
The main issue was fixed in release 1.6.0. @georgejcook, if you are still seeing issues with the vscode debugger output, please open a new issue
